exonum/src/runtime/types.rs

```diff
@@ -238,6 +258,25 @@ pub struct ArtifactSpec {
     pub payload: Vec<u8>,
 }
 
+impl ArtifactSpec {
+    /// Generic constructor.
+    pub fn new(artifact: ArtifactId, deploy_spec: impl BinaryValue) -> Self {
+        Self {
+            artifact,
+            payload: deploy_spec.into_bytes(),
+        }
+    }
+}
+
+impl<T: Into<ArtifactId>> From<T> for ArtifactSpec {
```

Comment:
This conversion seems to be very error-prone, since it hides the fact of setting payload to empty value. 
Service having no payload is not an invariant, but `From` conversion makes it feel like it is.

I suggest it to be a method with the name explaining it unambiguously, e.g. `new_without_spec` (and @slowli originally [suggested](https://github.com/exonum/exonum/pull/1541#discussion_r352093418) it to be a method, and his suggestion still makes sense).